### PR TITLE
refactor: deprecate util.find_git_ancestor

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/nvim-lspconfig/issues/2079.

### CODE REVIEW

File: lua/lspconfig/util.lua [hunk 1]
```diff
@@ -393,4 +383,9 @@ function M.find_package_json_ancestor(startpath)
   return vim.fs.dirname(vim.fs.find('package.json', { path = startpath, upward = true })[1])
 end
 
+--- @deprecated use `vim.fs.find('.git', { path = startpath, upward = true })[1]` instead
+function M.find_git_ancestor(startpath)
+  return vim.fs.find('.git', { path = startpath, upward = true })[1]
```
Comment (glepnir) [inline lua/lspconfig/util.lua:388]: use `vim.fs.dirname` to get parent dir of result  `{ "/Users/mw/workspace/neovim/.git" }`

Comment (justinmk) [reply lua/lspconfig/util.lua:388]: https://github.com/neovim/nvim-lspconfig/pull/3505

**Comment by dundargoc** [comment]
pinging @jacobrreed @folke to take a look in case I accidentally destroy the internet again

File: lua/lspconfig/util.lua (hunk 1 above)
Comment (glepnir) [reply lua/lspconfig/util.lua]: Considering this is a very common situation, maybe it is better to add parent = true to the filter table

**Review by justinmk — APPROVED** [review]

File: lua/lspconfig/util.lua (hunk 1 above)
Comment (dundargoc) [reply lua/lspconfig/util.lua]: You mean to vim.fs.find? But we already have vim.fs.root that does this, I don't think we need to add anything to vim.fs.find.

**Pull request merged**